### PR TITLE
Replace shutdown_room API with DELETE /_synapse/admin/v1/rooms/<room_…

### LINKED ISSUE #76: Replace shutdown_room and purge_room APIs with delete_room
Synapse is dropping the `shutdown_room` and `purge_room` APIs in favour of the `delete_room` API which handles both operations and more. Context: https://github.com/matrix-org/synapse/pull/8829

Please update your code to use this new endpoint instead :slight_smile:

### PR DESCRIPTION
**Pull request by Yoric**
…id> - Resolves #76, closes #96

As per https://github.com/matrix-org/synapse/issues/9052, shutdown_room is going away, to be replaced with DELETE /_synapse/admin/v1/rooms/<room_id>.

### CODE REVIEW

**Comment by Yoric** [comment]
Not clear how I can test it.

**Comment by richvdh** [comment]
@Yoric has promised we can get this deployed in the week of 2021/08/09.

File: src/Mjolnir.ts [hunk 1]
```diff
@@ -650,10 +650,12 @@ export class Mjolnir {
...
 
-    public async shutdownSynapseRoom(roomId: string): Promise<any> {
-        const endpoint = `/_synapse/admin/v1/shutdown_room/${roomId}`;
-        return await this.client.doRequest("POST", endpoint, null, {
+    public async shutdownSynapseRoom(roomId: string, message?: string): Promise<any> {
+        const endpoint = `/_synapse/admin/v1/rooms/${roomId}`;
+        return await this.client.doRequest("DELETE", endpoint, null, {
             new_room_user_id: await this.client.getUserId(),
+            block: true,
+            message: message || "This room was shutdown by a moderator"
```
Comment (turt2live) [inline src/Mjolnir.ts:658]: we should exclude the message so we can continue using the default

File: src/commands/ShutdownRoomCommand.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ limitations under the License.
 import { Mjolnir } from "../Mjolnir";
 import { RichReply } from "matrix-bot-sdk";
 
-// !mjolnir shutdown room <room>
+// !mjolnir shutdown room <room> [<message>]
```
Comment (turt2live) [inline src/commands/ShutdownRoomCommand.ts:20]: The help command also needs updating

File: package.json [hunk 3]
```diff
@@ -14,13 +14,15 @@
...
+    "@types/mocha": "^9.0.0",
+    "@types/node": "^16.4.5",
+    "eslint": "^7.31.0",
     "expect": "^27.0.6",
     "mocha": "^9.0.1",
     "ts-mocha": "^8.0.0",
     "tslint": "^6.1.3",
-    "typescript": "^4.3.5"
+    "typescript": "^4.3.5",
+    "typescript-formatter": "^7.2.2"
```
Comment (turt2live) [inline package.json:25]: These changes appear unrelated

**Review by turt2live — CHANGES_REQUESTED** [review]

File: src/Mjolnir.ts (hunk 1 above)
Comment (Yoric) [reply src/Mjolnir.ts:658]: I don't understand this comment. Should I remove `"This room was shutdown by a moderator"`?

Comment (turt2live) [reply src/Mjolnir.ts:658]: yea, just so the default message from Synapse can be used.

**Review by turt2live — APPROVED** [review]
